src/Controllers/AnnouncementController.php

```diff
@@ -104,4 +106,38 @@ public function edit(Request $request, $id)
             return ApiResponse::failed($e);
         }
     }
+
+    public function delete($id)
+    {
+        try {
+            $user = auth()->user();
+
+            $announcement = Announcement::join('badaso_courses', 'badaso_courses.id', '=', 'badaso_announcements.course_id')
+                ->where('badaso_announcements.id', $id)
+                ->where(function ($query) {
+                    $query->where('badaso_announcements.created_by', auth()->user()->id)
+                        ->orWhere('badaso_courses.created_by', auth()->user()->id);
+                })
+                ->select('badaso_announcements.*')
+                ->first();
+
+            if (! $announcement) {
+                throw ValidationException::withMessages([
+                    'id' => 'Announcement not found',
+                ]);
+            }
```

Comment:
if you see the Announcement model, there has been a function `course` which should be able to be called by `$announcement->course` and will return the corresponding course for the announcement. I think instead of putting the join logic here (which is also quite expensive), we can utilize that one.

```suggestion
            $announcement = Announcement::find($id);
            $course = $announcement->course;
            
            if (! ($announcement->created_by == $user->id || $course->created_by == $user->id) {
                throw ValidationException::withMessages([
                    'id' => 'Announcement not found',
                ]);
            }
```

Feel free to not implement it if you have another concern